torch/distributed/autograd/__init__.py

```diff
@@ -25,3 +27,34 @@ def __enter__(self):
 
     def __exit__(self, type, value, traceback):
         _release_context(self.autograd_context._context_id())
+
+
+def backward(roots):
+    '''
+    Kicks off the distributed backward pass using the provided roots. This
+    currently implements the "FAST" mode
+    (see https://github.com/pytorch/pytorch/issues/23110) algorithm which
+    assumes all RPC messages across workers would be part of the autograd graph
+    during the backward pass.
+
+    We use the provided roots to discover the autograd graph and compute
+    appropriate dependencies. This method blocks until the entire
+    autograd computation is done.
+
+    We accumulate the gradients in the appropriate "autograd context id" on each
+    of the nodes. The autograd context id used is the current autograd context
+    id of this node when backward() is called. If there is no valid autograd
+    context id, we throw an error.
+
+    Arguments:
+        roots: List of tensors which represent the roots of the autograd
+            computation. All the tensors should be scalars.
+
+    Example::
+        >> import torch.distributed.autograd as dist_autograd
+        >> with dist_autograd.context() as context_id:
```

Comment:
Do we need `as context_id` here? Can this be a thread_local var?

Reply:
We probably need to expose the `context_id` to the user since `get_gradients()` takes context_id as input.